include/zephyr/pm/device.h

```diff
@@ -718,7 +719,7 @@ static inline int pm_device_driver_init(const struct device *dev, pm_device_acti
 
 	/* When power management is not enabled, all drivers should initialise to active state */
 	rc = action_cb(dev, PM_DEVICE_ACTION_TURN_ON);
-	if (rc < 0) {
+	if ((rc < 0) && (rc != -ENOTSUP)) {
```

Comment:
Ah, regarding my comment above, I see that you changed it here to avoid adding that case in pm action function but i am not sure if that is ok. There might be other reasons why turn on action reports `-ENOTSUP` (e.g. turning on device in unsupported configuration).

Reply:
We could standardize on an error code to signal this condition, I thought `-ENOTSUP` is a reasonable choice.

Reply:
The pm action callback documents:
```
 * @retval -ENOTSUP If the requested action is not supported.
```
so there is definately precedent for it :)